src/configure/configure.ts

```diff
@@ -121,34 +122,45 @@ class Orchestrator {
             await this.getAzureSession();
             await this.getSelectedPipeline();
 
-            if (this.inputs.pipelineConfiguration.template.label === "Containerized application to AKS") {
-                // try to see if node corresponds to any parameter of selected pipeline.
-                if (resourceNode) {
-                    let resourceParam = TemplateParameterHelper.getMatchingAzureResourceTemplateParameter(resourceNode, this.inputs.pipelineConfiguration.template.parameters);
-                    if (resourceParam) {
-                        this.inputs.pipelineConfiguration.params[resourceParam.name] = resourceNode;
+            
+            if(this.inputs.pipelineConfiguration.templateInfo) {
```

Comment:
I had a discussion with @kanika1894 on this and I feel the code is getting cluttered due to a lot of if-else. Essentially we have three different ways of creating pipeConfiguration parameters here - one is the original one which did not use any templates (and was used in app service), second one is for AKS which used local templates and now we are adding a third one which will use remote templates.
We should use factory pattern here. The code is this class/function should be just a series of abstract functions. A factory should return the appropriate implementation of that functionality. And all the real logic will move to such implementation classes.

Reply:
As discussed offline this will be taken care of in separate PR by Kanika.